creator-node/src/services/sync/secondarySyncFromPrimary.js

```diff
@@ -49,14 +49,21 @@ const handleSyncFromPrimary = async ({
   /**
    * Perform all sync operations, catch and log error if thrown, and always release redis locks after.
    */
-  const forceResync = await shouldForceResync(
+  const shouldForceResyncBool = await shouldForceResync(
     { libs, logContext },
     forceResyncConfig
   )
+  const forceResyncQueryParam = forceResyncConfig?.forceResync
+  if (forceResyncQueryParam && !shouldForceResyncBool) {
+    return {
+      error: new Error(`Cannot issue sync for wallet ${wallet} due to shouldForceResync() rejection`),
+      result: 'failure_force_resync_check'
```

Comment:
I dont think we should throw here. what if a bad actor tried to force resync? or, if this was a replay attack and the timestamp expired? then the function would be acting correctly and rejecting the force resync. I think we need to be more specific in tracking failure, like if the recovering signature went well and force resync was enabled, but the user state clearing failed. cc @dmanjunath

Reply:
just to clarify this is early exiting from the entire sync because the forceResync flag was set but there was some issue verifying the force resync?

Reply:
correct @dmanjunath

Reply:
i think i understand what @SidSethi is doing here. basically if we continue on the sync, it should basically be fine but we don't know that the force resync verification failed, this is the best way to figure that out

Reply:
exactly

Reply:
not only is this the best way to figure out when that happened, but we also should not continue the sync

Reply:
yeah i think this is a reasonable approach. @vicky-g i think you're kind of saying the same thing too. even in your case if someone maliciously called forceResync we probably shouldn't continue or if there was a clock skew issue, they shouldn't be asking a node to force resync

Reply:
im not a fan of coupling the force resync logic with blocking the sync. we really have to make sure across the board that sync is called correctly if we're pushing this early exit

Reply:
i'm not sure i follow - this would only reject sync if forceResync query param is set and the verification fails. isn't that what we want?

Reply:
yeah i don't see this as "coupling" the logic. forceresync is a conditional part of sync with validation just like clock values being contiguous, malformed response etc. returning early here isn't a new thing we're adding for forceresync. i think this change is reasonable

Reply:
@vicky-g any concerns still?

Reply:
no, just gotta be careful that we're calling force resync correctly